### PR TITLE
[MRG+1] ENH: Corrects order and behavior of filters in `nilearn.signal.clean`

### PR DESCRIPTION
**Pull request by miykael**
This PR changes the order of the cleaning function, as discussed in https://github.com/nilearn/nilearn/issues/1828.

As a new feature, confound removal will be done orthogonally to temporal filters (according to [Lindquist et al. (2018)](https://www.biorxiv.org/content/biorxiv/early/2018/09/04/407676.full.pdf)), if both are specified.

### CODE REVIEW

**Comment by bthirion** [comment]
Also, I don't see travis/circleci output. Am I missing something ?

**Comment by miykael** [comment]
I think circleci and travis are coming back.

About the test. How should I best implement a test that checks if confound and band-pass filters work correctly?
My inital approach would be: Clean signal with confound and high pass specified and then check the power spectrum for the filtered frequency. If the filters are not orthogonal, then some filtered frequency should be reintroduced into the data. But I'm not sure if this is the smartest approach.

**Comment by GaelVaroquaux** [comment]
I think that it is important that this change is mentioned in the whats_new.

File: nilearn/signal.py [hunk 1]
```diff
@@ -508,11 +517,28 @@ def clean(signals, sessions=None, detrend=True, standardize=True,
...
         confounds = _ensure_float(confounds)
         confounds = _standardize(confounds, normalize=standardize,
                                  detrend=detrend)
+
+        # Apply low- and high-pass filters to keep filters orthogonal
+        # (according to Lindquist et al. (2018))
+        if low_pass is not None or high_pass is not None:
+
+            confounds = butterworth(confounds, sampling_rate=1. / t_r,
+                                    low_pass=low_pass, high_pass=high_pass)
```
Comment (GaelVaroquaux) [inline nilearn/signal.py]: We should filter before standardizing the confounds: the standardizing of confounds is useful for numerical stability.

**Comment by GaelVaroquaux** [comment]
Overall looks great aside two small comments.

**Comment by miykael** [comment]
Moved filtering before standardizing and added changes to whats_new.

**Comment by GaelVaroquaux** [comment]
Some of your jobs had failed for no good reasons (Internet flukes on the CI). I have restarted them.

**Comment by GaelVaroquaux** [comment]
> About the test. How should I best implement a test that checks if confound and band-pass filters work correctly?
> My inital approach would be: Clean signal with confound and high pass specified and then check the power spectrum for the filtered frequency. If the filters are not orthogonal, then some filtered frequency should be reintroduced into the data. But I'm not sure if this is the smartest approach.

I think that this is a good idea: it should be a reasonably simple test that checks that the code does what it should.

**Comment by GaelVaroquaux** [comment]
Also, as you add the test, can you merge with master, to resolve the conflicts that I added when merging your other branch.

Thanks a lot!

**Comment by miykael** [comment]
I've resolved the conflicts and added the tests (https://github.com/nilearn/nilearn/pull/1837/commits/7b445badebb0577ddbd420439020922e2bd02150). A detailed explanation/description of the tests can be found [here](https://gist.github.com/miykael/4bb887029cba3926902feb2e4afe190f).

File: doc/whats_new.rst [hunk 2]
```diff
@@ -5,14 +5,23 @@
...
+  is according to Lindquist et al. (2018).
+
 
 Contributors
 -------------
 
 The following people contributed to this release::
 
-  1  Michael Notter
+  2  Michael Notter
```
Comment (GaelVaroquaux) [inline doc/whats_new.rst]: I wouldn't worry about keeping this up to date: it's something that we do when we do the release.

**Comment by GaelVaroquaux** [comment]
LGTM. +1 for merge.

@bthirion : do you want to have another look?

**Comment by KamalakerDadi** [comment]
Is it such a normal procedure to do that we don't worry about documenting the details ?
For instance, in doc/manipulating_images/manipulating_images.rst

**Comment by GaelVaroquaux** [comment]
I don't think that we should document the details in another place than the function docstring, because the danger is that as the function evolves (for instance with addition of the DCT based filtering) the other documentation is not kept up to date.

**Comment by GaelVaroquaux** [comment]
Thanks heaps. Merging!

**Pull request merged**